rl.py

```diff
@@ -32,144 +33,281 @@
+    parser.add_argument("-a", "--algo", help = "RL algorithm",
+                        type=str, required=True)
+    parser.add_argument("-ne", "--env_num", help = "Number of parallel envs",
+                        type=int, required=False, default=4)
+    parser.add_argument("-s", "--seed", help = "Seed",
+                        type=int, required=False, default=0)
+    parser.add_argument("-ep", "--epochs", help = "Number of training epochs",
+                        type=int, required=False, default=100)
+    parser.add_argument("-st", "--steps", help = "Number of learning steps",
+                        type=int, required=False, default=100)
+    parser.add_argument("-b", "--batch", help = "Batch size",
+                        type=int, required=False, default=64)    
+    parser.add_argument("-lr", "--learn_rate", help = "Learning rate",
+                        type=float, required=False, default=1e-4)
+    parser.add_argument("-g", "--gamma", help = "Discount factor",
+                        type=float, required=False, default=0.99)
+    parser.add_argument("-tr", "--train", help = "Training a new model or continue training from saved model", 
+                        choices=['new', 'continue', 'none'], required=False, default='new')
+    parser.add_argument("-te", "--test", help = "Testing flag",
+                        action=argparse.BooleanOptionalAction, default=False)
+    parser.add_argument("-tn", "--num_test", help = "Number of tests",
+                        type=int, required=False, default=1)    
+    parser.add_argument("-md", "--model_dir", help = "Model directory",
+                        type=str, required=False, default=None)
+    parser.add_argument("-ms", "--model_step", help = "Model time step",
+                        type=str, required=False, default=None)
+    
     args = parser.parse_args()
 
-    ids = ['cartstemcontact-v0', 'cartstem-v0', "stempendulum-v0",
-            'catchtheobject-v0', "multigaitrobot-v0"]
-    timer_limits = [30, 80, 50, 30, 18]
-    continues = [True, False, True, True, False]
-    n_epochs = [600, 200, 10001, 200, 10001]
+    env_name = args.env
+    envs_values = envs.values()
+    if env_name not in envs_values:
+        print(Fore.RED + '[ERROR]   ' + Fore.RESET + "Environment name does not exist.")
+        raise SystemExit(f"Available environments:\n{[value for value in envs_values]}")
+
+    algo_name = args.algo
+    algo_values = algos.values()
+    if algo_name not in algo_values:
+        print(Fore.RED + '[ERROR]   ' + Fore.RESET + "Algorithm not available.")
+        raise SystemExit(f"Available algorithms:\n{[value for value in algo_values]}")
+
+    n_env = args.env_num
+    seed = args.seed
+    n_epochs = args.epochs
+    n_steps = args.steps
+    batch_size = args.batch
+    lr = args.learn_rate
+    gamma = args.gamma
+    TRAIN = args.train
+    TEST = args.test
+    n_test = args.num_test
+
+    layers_size = [512, 512, 512]
```

Comment:
It is usually not a good practice to use hard coded values. But it is maybe out of the scope of this PR

Reply:
I moved this part to yaml files to save all the parameters of the model and the training, they can be changed by editing the yaml files directly